python/cudf/cudf/utils/ioutils.py

```diff
@@ -948,7 +948,8 @@
     the column names: if no names are passed, header=0;
     if column names are passed explicitly, header=None.
 names : list of str, default None
-    List of column names to be used.
+    List of column names to be used. Needs to include names of all column in
+    the file, or names of all columns selected using `usecols` (indices only).
```

Comment:
indices only? pytest example has column names too.

Reply:
the requirement to have the same number of elements as `usecols` is limited to the case when `usecols` contains indices.
I included the failing cases in the C++ test `UseColsValidation`, the options object names indicate the reason why it should fail.
I can add comments there to make it clearer. Also, let me know if this comment should be worded differently to be clearer.

Reply:
`(indices only)` is only confusing. does `indices only`  mean `usecols` uses indices, but not column names?

Rest all looks good.

Reply:
expanded the related comments, I hope they make sense now.